src/Components/Web.JS/src/JSInitializers.ts

```diff
@@ -0,0 +1,53 @@
+import { IBlazor } from "./GlobalExports";
+
+type BeforeBlazorStartedCallback = (...args: unknown[]) => Promise<void>;
+export type AfterBlazorStartedCallback = (blazor: IBlazor) => Promise<void>;
+type BlazorInitializer = { beforeStart: BeforeBlazorStartedCallback, afterStarted: AfterBlazorStartedCallback };
+
+export class JSInitializer {
+  private afterStartedCallbacks: AfterBlazorStartedCallback[] = [];
+
+  async importInitializersAsync(
+    initializerFiles: string[],
+    initializerArguments: unknown[]): Promise<void> {
+
+    try {
+      await Promise.all(initializerFiles.map(f => importAndInvokeInitializer(this, f)));
+    } catch (error) {
+      console.warn(`A library initializer produced an error before starting: '${error}'`);
```

Comment:
What's the reason for swallowing the exception? Shouldn't we let initialization fail?

Reply:
Same with the other try/catch below.

Reply:
Fair enough, I guess folks need to be responsible for what they bring into their apps, and deal with it if it crashes their app.